src/usb.c

```diff
@@ -399,6 +399,19 @@ void usb_i2c_restore_irqs(const char *i2c_path, bool force)
 
 void usb_hpm_restore_irqs(bool force)
 {
+    /*
+     * Do not try to restore irqs on M3/M4 which don't use i2c
+     */
+    if (adt_path_offset(adt, "/arm-io/nub-spmi-a0/hpm0") > 0)
+        return;
```

Comment:
This starts growing to a level where we probably should start thinking of adding some sort of device abstraction like Sven did with CPU initialization and I started for display initialization.

Reply:
that would be a possibility as well